platform/server-api/src/main/java/org/motechproject/server/api/PomInformation.java

```diff
@@ -0,0 +1,162 @@
+        } catch (FileNotFoundException ex) {
+            LOGGER.error("Error while opening POM file", ex);
+        }
+    }
+
+    public void parsePom(InputStream inputStream) {
+        try {
+            MavenXpp3Reader reader = new MavenXpp3Reader();
+
+            Model model = reader.read(inputStream);
+
+            if (dependencies == null) {
+                dependencies = new LinkedList<>();
+            }
+            if (repositories == null) {
+                repositories = new LinkedList<>();
+            }
+
+            parent = model.getParent();
+            setPropertiesFromModel(this, model);
+
+            for (org.apache.maven.model.Dependency dependency : model.getDependencies()) {
+                if (!"test".equalsIgnoreCase(dependency.getScope())) {
+                    dependencies.add(new Dependency(new DefaultArtifact(
+                            (dependency.getGroupId().contains("${")) ? parent.getGroupId() : dependency.getGroupId(),
+                            dependency.getArtifactId(),
+                            dependency.getClassifier(),
+                            "jar",
+                            dependency.getVersion()
+                    ), JavaScopes.RUNTIME));
+                }
+            }
+
+            for (Repository remoteRepository : model.getRepositories()) {
+                repositories.add(new RemoteRepository(remoteRepository.getId(), "default", remoteRepository.getUrl()));
+            }
+        } catch (Exception e) {
+            LOGGER.error("Error while reading POM file", e);
+        }
+    }
+
+    public void parseParentPom(File file) {
+        try {
+            parseParentPom(new FileInputStream(file));
+        } catch (FileNotFoundException ex) {
+            LOGGER.error("Error while opening parent POM file", ex);
+        }
+    }
+
+    public void parseParentPom(InputStream inputStream) {
+        try {
+            MavenXpp3Reader reader = new MavenXpp3Reader();
+
+            Model model = reader.read(inputStream);
+
+            PomInformation parentPom = new PomInformation();
+
+            setPropertiesFromModel(parentPom, model);
+            parentPom.setParent(model.getParent());
+
+            this.parentPomInformation = parentPom;
+        } catch (Exception ex) {
+            LOGGER.error("Error while reading parent POM file", ex);
+        }
+    }
+
+    /**
+     * Sets properties from parsed pom file to the pomInformation object.
+     * Additionally this method adds project.version as property if the
+     * version tag is used in the pom file.
+     *
+     * @param pomInformation the information about parsed pom file
+     * @param model the model from parsed pom file
+     */
+    private void setPropertiesFromModel(PomInformation pomInformation, Model model) {
+        Properties propertiesFromModel = model.getProperties();
+
+        if (model.getVersion() != null) {
+            propertiesFromModel.put("project.version", model.getVersion());
```

Comment:
Regarding my earlier question, I see that it will work for ${project.version} since we are hardcoding it. I am assuming other maven variables will not work however?